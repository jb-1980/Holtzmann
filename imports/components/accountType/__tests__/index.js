import renderer from "react-test-renderer";
import AccountType from "../index";

it("should exist", () => {
  const accountType = renderer.create(<AccountType type="American Express" />);
  expect(accountType).toMatchSnapshot();

  const ach = renderer.create(<AccountType type="ACH" />);
  expect(ach).toMatchSnapshot();
});

it ('should show different icons for AmEx and ACH', () => {
  const amEx = renderer.create(<AccountType type="American Express" />);
  const ach = renderer.create(<AccountType type="ACH" />);

  const amExJSON = amEx.toJSON();
  const achJSON = ach.toJSON();

  expect(amExJSON).not.toBe(achJSON);
});

it ('should allow custom height and width', () => {
  const accountType = renderer.create(<AccountType width="105" height="106" type="ACH" />);
  const json = accountType.toJSON();

  expect(json.props.style.width).toBe('105');
  expect(json.props.style.height).toBe('106');

  // TODO: fails for reason https://github.com/NewSpring/Holtzmann/issues/1133
  // expect(json.props.style.marginBottom).toBe('-10.6px');
});
