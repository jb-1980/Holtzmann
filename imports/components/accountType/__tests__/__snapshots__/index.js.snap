exports[`test should exist 1`] = `
<div
  className="background--fill display-inline-block"
  style={
    Object {
      "backgroundImage": "url(\'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFAAAAA8CAYAAADxJz2MAAAABGdBTUEAALGPC/xhBQAABpBJREFUeAHtnGtsFFUUgM/strRlW9rad9NHoAV5PwRB5WFsUQQUCRhNiVWRR0uCGE1MNDEKBklsAiEmJlIgIQEkiEGCKCCWKCgo0GKUFvsAStpuWwotfUGfO54zZWZnZvfOXjrxV+9JZ+/ce8+5d+a7554zzW5Gggciy7I0YmfJao8sr5JkmCCDHK72iRJAAqlDlqDUIUm729Y8tkuSkBKKRB+xOy4ldcmwD6FlUV2INQFJgqJQkHJv582od5DnCXjWwMy9sgzZXSDvJXZSRGHxGo/HU2hWEvXABBwOx1oHxbzAqkLDHwFi56CE4a9TtAUmQOwcItsGBsXSIHYOVqdo5yMgAPJxYmoJgEw0fB0CIB8nppYAyETD1yEA8nFiagmATDR8HQIgHyemlgDIRMPXIQDycWJqCYBMNHwdAiAfJ6aWAMhEw9chAPJxYmoJgEw0fB0CIB8nppYAyETD1xHEpzag9eXT6bBidIyPSWnzfZhz+KpPe+QwJ1S/PsWn3YMtsw6VQlVrt0/f1tlp8Na4WJ/2fRV34NuqZjiyaLRPn1XDiz9UKPOcXPIopIUP01R7PDIsO14Jv9d3aG10sjtrJCwbFW1o++LvRvjkQp2hTa1we2CyK1iBF+x0gPmYGueCJSOj1DENpVmX6iF4vDs10aBHlfiwIHhjbKzP+GTjxC9j8c9vn7851Da0goZ7vbAcYbX09Gv2rmAnHFyQCeOiQ7Xr+OyJFHgVHUS1pfJAZTMTHhlyA3x7coIysDab6eTD6cmmFutqDl4oLYpeaI7QIO5L0psGPCdvfwm9sbW7T9ONCgmC7xaNgaThwbBuYjy8M8W4qEeut8D6M9Wavr8Tri0cHeLEbRVnsN9ZeguWZTwCMaEDQ0yKGa544dEbdw16+kpDZw8kuga20TBc3Q2TE+GD8zWKCm331ePjNXW9rtZoOtl4oRa2Xm4wtRqryu8vHjT9c+c+emIVHF08GoajB5Kk4LYuWjpWKR+oKUVRbSusLLoOuNMthQtg3oR4IJdXpbO3HzZfckNtRw9smpWiNgN5oRXA840dCvB5ySMUm5UY6wpK3NDc3Q956AERCFGVgsv1sG1Oulr1W9I4QbSvLWQ/xs4avE5V/sBrWHHqGhzC7UtblCQtIkTtVso/Gzog5+Q16A1ED7UDAgzDLZWPN6eXQvS+O119sAPLDej2D+OFGzEYn146AJAWZd2kBNj2V4OyhdQ56AaO32xFgGqL/zIrZQTQYSWUJPQASffnmjYlrm15MtXHlLb4yycq4V4fpbrAEjDgvIlBPTbMG6vw9yBQ0nQP5iZFwLRYF5yubTPMEigWXmjshBM3vducFmc9QozTzUGQ/0+JQk9fNd4YktT5IjEursUdxyuWHujE3bEBA7te8GddsPfZDH2T4ZwnFm66WAcL0iIxq0oQjRf88ePeBEQLcra+HVJ1jxyGCXQV0j2HulZS3W58VKJ7ouvPiPRmX7P9RzOSoQwfzb6v9i60WUetWwJ8JTMGUk3xQTW0KgPFQgrmhzHDLcckREIgVdn0EN53xt32UEmE5vj8qVR4xrTtD2CcpFCQgNmYhK5nFz4PZh/5F64gSCuxBPjeNGNatxpI36d64a91bO/YjF64dGQ0OB1eeMeqW6C4qVMZKkDyU3Q2zkwBOqxk4dFyxaNJh5JW/kTjjvrN3Q75v9yAmQnh8OMLY7TEQvH5m+czYR7+g3Ab4z1LmAAXpUfiQ2aYZkeZlwZrx9KfrMTHHH38o/Oz7nJ/qkpbJT6XfY0rn4sxlgR/6QSf6rzPi1Xptv0xOykctuF/OXqhp4hczMj9uFrnMXG9f64Gts/1Zn7Kzvufy4DFxyqgj5GRmUmEntF6+z3asePKLSi/2wXuzl6/x3bMpI34xK/ajI0KhYXpUVqd2s2JbUuxG2hhqO9gZTOUtXRp90ceqI5FZT8Cxj9Dm76fdY5WyjPenuxRytiqXgfOm/NTFTTpvGtXWRPsudpkmGNmvAsKcNuzRHJ9dZGuVcggCTA9cJDjDTkzAdDmkguAAqBNAjbNhQcKgDYJ2DQXHigA2iRg01x4oABok4BNc+GBAqBNAjbNhQcKgDYJ2DQXHigA2iRg01x4oF2A9DYKm2MMWXNi56BXeQxZAjZvXHkNCr0HxeY4Q9ac2En06o6Iwkun8Buv7CFLYhA3jtv3dHve9PkIUZLpJTL4ZXzRIMYZkiYEL1SC14id9v01eaJ49RPbHyhh+Hv103/4PGN0rRTg3wAAAABJRU5ErkJggg==\')",
      "borderRadius": "3px",
      "height": 40,
      "position": "inherit",
      "top": "3px",
      "width": 54
    }
  } />
`;

exports[`test should exist 2`] = `
<div
  className="background--fill display-inline-block"
  style={
    Object {
      "backgroundImage": "url(\'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFAAAAA8CAYAAADxJz2MAAAABGdBTUEAALGPC/xhBQAAAc9JREFUeAHt3DFOwzAUxnGcROoMEx04Q4/AoSraoXMWqp4DjgFH4AwgdWvnDlV4FoyOn+xPlVrlnwEJ+32h/eWlcqjkcGfHbrd7Op1Or8MwPNuv8zjGMSqwDyF8tG27Wq/XP+Ef78vw7kcjTKQEDl3XLRrrvC14KR937OF8Pm8bK4u3LUeFQPzIa+zHY0WWyJ/APHYghyDQjWU3m00Ym5vieN/3Q+p904EplYIxAAuwUqUAplQKxgAswEqVAphSKRgDsAArVQpgSqVgDMACrFQpgCmVgrHRJxHvHGMrcy93rfO1T150oHhFAQRQFBDjdCCAooAYpwMBFAXEePU6sHbdJL7eq4tzC4uXBEAARQExTgcCKAqIcToQQFFAjNOBAIoCYrz6SeTS/5G+lScdbmGxAwEEUBQQ43QggKKAGKcDARQFxHj1OvBW1mmijxvnFnaJ8gUA5n3cWQBdonwBgHkfdxZAlyhfAGDex50F0CXKFwCY93FnAXSJ8gUA5n3cWQBdonxBuPR3G/k/f/uzdKB4DSPgXjzHlOP7xjaR+ZyygPLeo11jO/C82EkOyommmDW842w2WzVx+6K4A48NvBkEt7PfDXHrp3fDWyyXy+9ftplH0FR79EYAAAAASUVORK5CYII=\')",
      "borderRadius": "3px",
      "height": 40,
      "position": "inherit",
      "top": "3px",
      "width": 54
    }
  } />
`;
